### PR TITLE
More gdb introspection on CI

### PR DESCRIPTION
**Pull request by azat**
Changelog category (leave one):
- Not for changelog (changelog entry is not required)

This is required to get more details for things like from:
- #31531
- #33057

```
Failed assertion: "bin->low_bits_full > bin->low_bits_empty" Received signal 6 Received signal Aborted (6)
```

Cc: @tavplubix

### CODE REVIEW

File: docker/test/fuzzer/run-fuzzer.sh [hunk 1]
```diff
@@ -399,3 +398,10 @@ EOF
 esac
 
 exit $task_exit_code
+
+# Core dumps (see gcore)
+# Default filename is 'core.PROCESS_ID'
+for core in core.*; do
+    pigz $core
+    mv $core.gz /output/
+done
```
Comment (azat) [inline docker/test/fuzzer/run-fuzzer.sh:407]: By the way, I'm not sure that artifacts via `/output` works for `fuzzer`, is this correct?

Comment (tavplubix) [reply docker/test/fuzzer/run-fuzzer.sh:407]: It's not: https://github.com/ClickHouse/ClickHouse/blob/73a593b67625aa9b478594f2f5d675e215347e25/tests/ci/ast_fuzzer_check.py#L95-L108

Comment (azat) [reply docker/test/fuzzer/run-fuzzer.sh]: Added uploading `core.gz`, and here is a worked example - https://s3.amazonaws.com/clickhouse-test-reports/33057/c43709a12fcca4452331464d31ace4df804e8796/fuzzer_astfuzzerdebug,actions//report.html

**Comment by azat** [comment]
There was one issue with the report for fuzzer, now everything should be correct.

**Review by tavplubix — APPROVED** [review]

**Pull request merged**